### ISSUE TITLE
[MPS] F.conv1d and F.conv2d produce incorrect gradients when minibatch >= 2^16

### ISSUE THREAD

**Issue by lanmelo** [pytorch/pytorch#96225]
### 🐛 Describe the bug

Both `F.conv1d` and `F.conv2d` (and `torch.nn.Conv1d` and `torch.nn.Conv2d`) produce incorrect gradients
when the minibatch size is 65_536 (=2^16) or greater. In the example below, the bottom two assertions fail.
```python
import torch

def test_grad(minibatch, dimensions=1):
    if dimensions == 1:
        view = (-1, 1, 1)
        op = torch.nn.functional.conv1d
    elif dimensions == 2:
        view = (-1, 1, 1, 1)
        op = torch.nn.functional.conv2d

    # create identical input tensors
    input_cpu = torch.ones(minibatch)
    input_mps = copy.deepcopy(input_cpu).to("mps")

    # create identical weight tensors
    weight_cpu = torch.ones(1)
    weight_mps = copy.deepcopy(weight_cpu).to("mps")
    weight_cpu.requires_grad_()
    weight_mps.requires_grad_()

    # get output
    out_cpu = op(input_cpu.view(view), weight_cpu.view(view)).mean()
    out_mps = op(input_mps.view(view), weight_mps.view(view)).mean()
    out_cpu.backward()
    out_mps.backward()

    # compare CPU and MPS outputs
    assert torch.allclose(out_cpu, out_mps.to("cpu"))
    assert torch.allclose(weight_cpu.grad, weight_mps.grad.to("cpu")) # fails with minibatch >= 65_536

test_grad(65_535, dimensions=1)
test_grad(65_535, dimensions=2)
test_grad(65_536, dimensions=1) # second assertion fails
test_grad(65_536, dimensions=2) # second assertion fails
```

The output might be non-deterministic, but in the testing below the error rate with `minibatch=65_535` is 6%,
whereas the error rate with `minibatch=65_536` is 100%.
```python
n_iter = 100
errors_65_535 = 0
errors_65_536 = 0

for _ in range(n_iter):
    try:
        test_grad(65_535)
    except AssertionError:
        errors_65_535 += 1
    
    try:
        test_grad(65_536)
    except AssertionError:
        errors_65_536 += 1

print(f"65_535 AssertionError rate = {errors_65_535 / n_iter : .0%}")
print(f"65_536 AssertionError rate = {errors_65_536 / n_iter : .0%}")
```

### Versions

PyTorch version: 1.13.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.2.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.7
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.6 (main, Aug 29 2022, 10:06:59) [Clang 13.1.6 (clang-1316.0.21.2.5)] (64-bit runtime)
Python platform: macOS-13.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Max

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.24.2
[pip3] torch==1.13.1
[pip3] torchtext==0.13.1
[conda] Could not collect
